packages/material-ui/src/Switch/Switch.js

```diff
@@ -121,11 +123,37 @@ export const styles = theme => ({
       theme.palette.type === 'light' ? theme.palette.common.black : theme.palette.common.white,
     opacity: theme.palette.type === 'light' ? 0.38 : 0.3,
   },
+  /* Small size */
+  sizeSmall: {
```

Comment:
An alternative proposal idea:
```jsx
/* Small size */
sizeSmall: {
  width: 30 + 10 * 2,
  height: 10 + 10 * 2,
  padding: 10,
  '& $thumb': {
    width: 18,
    height: 18,
  },
  '& $switchBase': {
    padding: 6,
  },
},
```

What do you think?

Reply:
Thank you for this tip!

Reply:
Oh, I was only proposing the idea. I was curious to hear your pros/cons of it. The win of your original is that it doesn't increase specificity, it makes it easier for somebody to override the styles. The win of this proposal is that it doesn't require the creation of new style rule. It can also be reverted easily back to the original approach (the inverse is not possible). Hum, ok, I think that we can stay this way using the last argument: the switching cost is low. We can revert back to your initial proposal in the future without having to worry about breaking changes.

Reply:
Actually, I think this is a more convenient approach in this case. If you want to customize the size of the switch you have to adjust the 3 elements - root, thumb and switch base. So, you can do it in one place.